configs/components/rubygem-ffi.rb

```diff
@@ -66,7 +66,8 @@
   pkg.environment "MAKE", platform[:make] if platform.is_solaris?
 
   if platform.is_cross_compiled_linux? || platform.is_solaris?
-    pkg.environment "PATH", "/opt/pl-build-tools/bin:/opt/csw/bin:$(PATH)"
+    # PA-5639, if we back out the CSW GCC change we should remove the CSW from this path
+    pkg.environment "PATH", "/opt/csw/bin:/opt/pl-build-tools/bin:$(PATH)"
```

Comment:
This ends up affecting `agent-runtime-7.x` too:

```
$ bx rake vanagon:component_diff -- -P agent-runtime-7.x -p all --from 1a77733 --to HEAD  | grep 'Platform name'
Platform name: solaris-11-sparc
Platform name: sles-12-ppc64le
Platform name: el-7-ppc64le
Platform name: el-7-ppc64
Platform name: solaris-11-i386
Platform name: solaris-10-sparc
Platform name: solaris-10-i386
``` 

All of the `ppc*` platforms can be ignored, but I'm not sure if changing the path will break 7.x builds, for example:

```
Platform name: solaris-10-i386
    Component 'rubygem-ffi'
        Field: environment
        --------------------
        - GEM_HOME=/opt/puppetlabs/puppet/lib/ruby/gems/2.7.0 GEM_PATH=/opt/puppetlabs/puppet/lib/ruby/gems/2.7.0 RUBYLIB=/opt/puppetlabs/puppet/lib/ruby/vendor_ruby:$(RUBYLIB) MAKE=/opt/csw/bin/gmake PATH=/opt/pl-build-tools/bin:/opt/csw/bin:$(PATH)
        + GEM_HOME=/opt/puppetlabs/puppet/lib/ruby/gems/2.7.0 GEM_PATH=/opt/puppetlabs/puppet/lib/ruby/gems/2.7.0 RUBYLIB=/opt/puppetlabs/puppet/lib/ruby/vendor_ruby:$(RUBYLIB) MAKE=/opt/csw/bin/gmake PATH=/opt/csw/bin:/opt/pl-build-tools/bin:$(PATH)
```

Maybe make this change conditionally `if settings[:runtime_project] == 'agent-runtime-main'`?

Reply:
Added in the `agent-runtime-main` logic.